ath79: expand rootfs for DIR-825-B1 with unused space

Lucky1openwrt opened the pull request:
ath79: expand rootfs for DIR-825-B1 with unused space

Expand currently unused flash space to roofs for DIR-825-B1 by using the same
flash space as the old ar71xx big image without moving the caldata.

With some testing this partition is use in the OEM firmware
but if changed is regenerated which allows reverting to OEM firmware.

Signed-off-by: Alan Luck <luckyhome2008@gmail.com>

adschm commented:
"room" -> "space"

Lucky1openwrt commented:
I think a part was used for the OEM saved config
but recreated if you ever factory default the device
the same area was also used in the old ar71xx fat image
dir-825-b1-fat-squashfs-sysupgrade.bin
tho stated to have a copy of cal data 
both of mine units didn't have this until it was put there
by the previously stated fat image version
this is just a way of keeping these end of life devices useful for a bit longer

Lucky1openwrt commented:
just a note
the flashable image size constraints remain so the image size can't change
this is mainly for version 2102 to add room after flashing initial image file
as by 22xx or 23xx the kernel+luci image size will most likely grow to big
to be able to flash in the room allowed via the 
old openwrt factory image needed to step to this one

adschm commented:
The "fat" images were usually meant for setups that should _not_ become default, i.e. when you have to adjust boot settings or alter stuff in a way that you cannot go back to OEM firmware. So, the fact that this setup corresponds to a "fat" image is actually a good reason _not_ to accept it as default for this device.

Lucky1openwrt commented:
the fat images would return a copy of the cal data to the correct spot
as long as you sysupgraded out of it 
& as noted is says to backup your caldata while installing "if you have to debrich out of it"
this only uses the same unused flash area that it did but without moving the cal data around
and you can go back to the OEM firmware without problem
as long as you reset the OEM firmware after flashing the factory config is recreated 

I don't under stand why you want to make these devices end there life earlier they you need to be

adschm commented:
Well, why don't you add a commit message then where you describe the situation and why your change is valid and helpful?

And please, try to use punctuation and capitalization. It's very hard to understand what you want to say.

Lucky1openwrt commented:
it that wording better ?

motolav commented:
You're editing the title of the PR not the commit message, your commit message is still just your signed off by.

Lucky1openwrt commented:
ok thanks dome now and added more info

adschm commented:
Apart from that, the relevant question is "Is it okay to overwrite the additional space?". This should be answered in the commit message.

lnjamn commented:
o

adschm commented:
> I took some time out to do some digging my old flash dumps do have some data part way into that partition 1 of the 2 units also had a DD_WRT config data in one of them I must have used DD-WRT as some pint before I created the DUMP
> 
> I can't tell what the data is & no text to reference it
> 
> I erased the unknown Partition on a unit Installed the oldest OEM firmware I could find V2.00 upgraded it a v2.05 and the last V2.10 here is a capture of v2.10's boot log around MTD allocation
> 
> ```
> 5 cmdlinepart partitions found on MTD device ar7100-nor0
> Creating 5 MTD partitions on "ar7100-nor0":
> 0x00000000-0x00040000 : "uboot"
> 0x00040000-0x00050000 : "Config"
> 0x00050000-0x00150000 : "vmlinux"
> 0x00150000-0x00660000 : "rootfs"
> 0x00660000-0x00670000 : "caldata"
> GACT probability on
> ```
> 
> note there is no MTD allocated to that area

Okay, so I would assume that this flash area is just entirely unused. But ...

> 
> then flashed OpenErt 18.06.9 factory image then SysUpgraded to 21.02.0
> 
> backed up & compared the partition there was some new data created in the partition the location & content was different to the old dump

... this tells us that somebody is writing to this area nevertheless.
But none of us know when this happened?

> 
> I could not see any change in the OEM firmware while using it It all seamed normal
> 
> I can't say 100% that erasing this partition has no effect
> 
> but using the ar71xx BIG image would have already erase what was there for those users

That's not our concern. That's why we offered a normal image, so those using the "big" image can do that on their own risk.

Honestly, I'm not really sure what's the best way to go on here. We know that flashing back OEM at least works in principle from your tests; it's not very probably that anybody will really want to use the OEM firmware in operation anymore anyway, except for flashing something different over it again. So maybe using that space would be fine after all ...

Anyway, before we can do anything, please finally update your commit, do the adjustments I asked for and give it a proper commit description summarizing the findings you have made.

Lucky1openwrt commented:
I have updated the commit message

but would love to hear other peoples opinions about this topic 
and see if it's wanted or worth pursuing in some way
I don't think anyone would be returning to the OEM firmware but who knows

robho commented:
> but would love to hear other peoples opinions about this topic

I'm not sure you want my opinion since I'm just a simple DIR-825-B1 user and not an openwrt developer, but I think this looks interesting. Thanks for creating this PR!

If I understand this correctly I would get ~1.6 MB extra space and that would be nice to have and make the device more usable to me.

Lucky1openwrt commented:
you may be loosing something from the OEM firmware if you ever decide to go back to it
to me the OEM firmware is so old & out dated I can't see me ever going back to it
but thanks for commenting :)

usver1 reviewed (approved):
Thank you for this PR. I've applied this patch and successfully build an flash a new image. Reclaiming almost 1/4 of flash space is a big deal for this old device.

Lucky1openwrt commented:
Thanks usver1

Goodwu reviewed (approved):
Compiled with this patch and works good.

Lucky1openwrt commented:
wow I did this as a bug fix for v21.02.rc's version hoping to get it into 21.02.0
look's like it has will miss v22 as well if ever

APCCV reviewed (approved):
Tested working great.
I have been using DIR-825 fat images for ages. The concern about overwriting the "unknown" partition is unfounded as that is what the fat image did. I moved back and forth from fat images to normal images many times, and back to stock many times, and never experienced issues.
This is a very elegant solution to make max flash space usable, much better than the discontinued fat images.

With a standard 21.02.3 build:
/dev/mtdblock5          960.0K    240.0K    720.0K  25% /overlay

With this patch, same build:
/dev/mtdblock8            2.5M    240.0K      2.3M   9% /overlay

That's 1.6M of extra free space by fully using all flash, available for user packages or as room for kernel/firmware to grow.

WreckTangles reviewed (approved):
Works. Used [this](https://github.com/WreckTangles/DIR825-21.02.3-concatenated/blob/DIR825-21.02.3-concatenated/bin/targets/ath79/generic/openwrt-21.02.3-ath79-generic-dlink_dir-825-b1-squashfs-sysupgrade.bin) image.

KMyrick reviewed (approved):
looks good

KMyrick reviewed (approved):
looks good, works.

Lucky1openwrt commented:
?

Lucky1openwrt commented:
?

Ansuel commented:
@Lucky1openwrt can you refactor the commit title and description and add some comments about the finding in the dts?

I would be good to have a clear commit description with understandable english (sorry if i sound rude). Currently it seems just a list of info that only some that knows about the topic can understand.

@adschm is 4 tester enough to accept this?

Lucky1openwrt commented:
ok thanks in a few hours I'll fix all this up 3am atm :)

Lucky1openwrt committed:
ath79: expand rootfs for DIR-825-B1 with unused space

Expand currently unused flash space to roofs for DIR-825-B1 by using the same
flash space as the old ar71xx big image without moving the caldata.

With some testing this partition is use by the OEM firmware
but if changed is regenerated which allows reverting to OEM firmware

Signed-off-by: Alan Luck <luckyhome2008@gmail.com>

Lucky1openwrt commented:
I hope this is suitable

DragonBluep commented on target/linux/ath79/dts/ar7161_dlink_dir-825-b1.dts:
```diff
@@ -198,10 +214,9 @@
+			fwconcat1: partition@670000 {
+				label = "fwconcat1";
 				reg = <0x670000 0x190000>;
-				read-only;
```
The final 64kb is a backup of `caldata` partition, It's better to keep it. Please refer to https://github.com/openwrt/openwrt/pull/9401

Lucky1openwrt replied:
this is cal data is only a product of using the ar71xx BIG
none of  dir-825-b1 devices had a copy of cal data there

DragonBluep replied:
Thanks for your explanation, I get it. https://github.com/openwrt/openwrt/commit/ffc0ea6c5d1ffe41a794926ba3b4a794f9833406

hauke commented:
Thank you for your patch, I applied it to master in aca8bb5cc332f0ffdf4249e76b0a56716f98bef0.
